### PR TITLE
Add feature to display the number of projects in each section

### PR DESCRIPTION
**Pull request by sparshalc**
Implementing a new feature to enhance the user experience by displaying the count of projects in each section. This feature addresses issue #4331.

image

### CODE REVIEW

**Review by KevinMulhern — CHANGES_REQUESTED** [review]

File: db/seeds.rb [hunk 1]
```diff
@@ -22,3 +22,8 @@
 
 # GENERATE accounts with admin privileges for testing
 load './db/seeds/test_admins.rb'
+
+# UPDATES projects_count and lessons_count after db is seeded
+Course.find_each do |course|
```
[the remark this answers is not in the document]
Comment (sparshalc) [reply db/seeds.rb:27]: Is /theodinproject/db/fixtures/paths/full_stack_rails/courses/databases.rb where you intend to put it?

[the remark this answers is not in the document]
Comment (sparshalc) [reply db/seeds.rb:27]: Is the intended location for it to be theodinproject/lib/seeds/course_builder.rb?

[the remark this answers is not in the document]
Comment (KevinMulhern) [reply db/seeds.rb:27]: We could put it there, but we'll also have to call it in all course files as an after action like we do with the [`destroy_removed_seeds`](https://github.com/TheOdinProject/theodinproject/blob/main/db/fixtures/paths/full_stack_javascript/seed.rb#L24) method.

I think for now, following the same format as our success stories seeds should be good enough. We'll just update all course counts in one go -  we'll need to put it in this location: `db/seeds/update_course_counts.rb`

File: app/views/courses/_course.html.erb [hunk 2]
```diff
@@ -14,7 +14,18 @@
           <h2 id="<%= course.title.parameterize %>" class="font-medium text-xl md:text-left text-gray-900 dark:text-gray-200">
             <%= course.title %>
           </h2>
-          <span class=" text-sm text-gray-500"><%= course.lessons.size %> Lessons</span>
+          <div class="flex space-x-3 mt-2 sm:mt-1.5 sm:justify-center text-gray-500 dark:text-gray-400">
+            <div class="flex items-center space-x-1">
+              <%= inline_svg 'icons/book.svg', alt: 'lesson icon', class: 'w-6 h-6' %>
+
+              <span class="text-sm text-gray-500"><%= pluralize(course.lessons_count, 'Lesson') %></span>
```
Comment (KevinMulhern) [inline app/views/courses/_course.html.erb:21]: Nit: Icon and text have different colors on dark mode, but we can inherit the color for both from the parent:

```suggestion
              <span class="text-sm "><%= pluralize(course.lessons_count, 'Lesson') %></span>
```

File: app/views/courses/_course.html.erb [hunk 3]
```diff
@@ -14,7 +14,18 @@
...
+            </div>
+            <div class="flex items-center space-x-1">
+              <%= inline_svg 'icons/wrench-screwdriver.svg', alt: 'lesson icon', class: 'w-6 h-6' %>
+
+              <span class="text-sm text-gray-500"><%= pluralize(course.projects_count, 'Project') %></span>
```
Comment (KevinMulhern) [inline app/views/courses/_course.html.erb:26]: Same here:

```suggestion
              <span class="text-sm"><%= pluralize(course.projects_count, 'Project') %></span>
```

File: db/seeds.rb [hunk 4]
```diff
@@ -22,3 +22,6 @@
 
 # GENERATE accounts with admin privileges for testing
 load './db/seeds/test_admins.rb'
+
+# GENERATE COURSE Content
```
Comment (KevinMulhern) [inline db/seeds.rb:26]:
```suggestion
# GENERATE course lesson and project counts
```

File: app/views/courses/_course.html.erb [hunk 5]
```diff
@@ -14,7 +14,18 @@
           <h2 id="<%= course.title.parameterize %>" class="font-medium text-xl md:text-left text-gray-900 dark:text-gray-200">
             <%= course.title %>
           </h2>
-          <span class=" text-sm text-gray-500"><%= course.lessons.size %> Lessons</span>
+          <div class="flex space-x-3 mt-2 sm:mt-1.5 sm:justify-center text-gray-500 dark:text-gray-400">
```
Comment (KevinMulhern) [inline app/views/courses/_course.html.erb:17]: Nit: From testing, it looks like we don't need the sm: breakpoint on justify-center

```suggestion
          <div class="flex space-x-3 mt-2 sm:mt-1.5 justify-center text-gray-500 dark:text-gray-400">
```

**Review by KevinMulhern — CHANGES_REQUESTED** [review]

File: app/views/courses/_course.html.erb [hunk 6]
```diff
@@ -14,7 +14,18 @@
           <h2 id="<%= course.title.parameterize %>" class="font-medium text-xl md:text-left text-gray-900 dark:text-gray-200">
             <%= course.title %>
           </h2>
-          <span class=" text-sm text-gray-500"><%= course.lessons.size %> Lessons</span>
+          <div class="flex space-x-3 mt-2 sm:mt-1.5 justify-center text-gray-500 dark:text-gray-400">
+            <div class="flex items-center space-x-1">
+              <%= inline_svg 'icons/book.svg', alt: 'lesson icon', class: 'w-6 h-6' %>
```
Comment (KevinMulhern) [inline app/views/courses/_course.html.erb:19]: Final pass through - the icons are just a little too big and draw the eye too much. 5 seems to make them more proportionate to their surroundings. 

```suggestion
              <%= inline_svg 'icons/book.svg', alt: 'lesson icon', class: 'w-5 h-5' %>
```

Size difference can be seen with this, first course has the smaller icons and the second doesnt.
Screenshot 2024-01-24 at 14 13 09

**Review by KevinMulhern — CHANGES_REQUESTED** [review]
Thanks for making those changes, just one more small change and I think we're good to ship. We'll need the merge conflict fixed too.

**Review by KevinMulhern — APPROVED** [review]
Looking great, thanks @sparshalc

**Pull request merged**